scipy/ndimage/tests/test_interpolation.py

```diff
@@ -460,72 +563,85 @@ def mapping(x):
         idx -= 1
+        idx = xp.asarray(idx)
 
         out = ndimage.map_coordinates(data, idx, order=order)
         assert_array_almost_equal(out, expected)
 
     @pytest.mark.parametrize('order', range(0, 6))
-    def test_map_coordinates02(self, order):
-        data = np.array([[4, 1, 3, 2],
-                         [7, 6, 8, 5],
-                         [3, 5, 3, 6]])
+    def test_map_coordinates02(self, order, xp):
+        data = xp.asarray([[4, 1, 3, 2],
+                           [7, 6, 8, 5],
+                           [3, 5, 3, 6]])
         idx = np.indices(data.shape, np.float64)
         idx -= 0.5
+        idx = xp.asarray(idx)
 
         out1 = ndimage.shift(data, 0.5, order=order)
         out2 = ndimage.map_coordinates(data, idx, order=order)
         assert_array_almost_equal(out1, out2)
 
-    def test_map_coordinates03(self):
-        data = np.array([[4, 1, 3, 2],
+    def test_map_coordinates03(self, xp):
+        data = _asarray([[4, 1, 3, 2],
                          [7, 6, 8, 5],
-                         [3, 5, 3, 6]], order='F')
+                         [3, 5, 3, 6]], order='F', xp=xp)
         idx = np.indices(data.shape) - 1
+        idx = xp.asarray(idx)
         out = ndimage.map_coordinates(data, idx)
-        assert_array_almost_equal(out, [[0, 0, 0, 0],
-                                        [0, 4, 1, 3],
-                                        [0, 7, 6, 8]])
+        expected = xp.asarray([[0, 0, 0, 0],
+                               [0, 4, 1, 3],
+                               [0, 7, 6, 8]])
+        assert_array_almost_equal(out, expected)
         assert_array_almost_equal(out, ndimage.shift(data, (1, 1)))
-        idx = np.indices(data[::2].shape) - 1
-        out = ndimage.map_coordinates(data[::2], idx)
-        assert_array_almost_equal(out, [[0, 0, 0, 0],
-                                        [0, 4, 1, 3]])
-        assert_array_almost_equal(out, ndimage.shift(data[::2], (1, 1)))
+
+        idx = np.indices(data[::2, ...].shape) - 1
+        idx = xp.asarray(idx)
+        out = ndimage.map_coordinates(data[::2, ...], idx)
+        assert_array_almost_equal(out, xp.asarray([[0, 0, 0, 0],
+                                                   [0, 4, 1, 3]]))
+        assert_array_almost_equal(out, ndimage.shift(data[::2, ...], (1, 1)))
+
         idx = np.indices(data[:, ::2].shape) - 1
+        idx = xp.asarray(idx)
         out = ndimage.map_coordinates(data[:, ::2], idx)
-        assert_array_almost_equal(out, [[0, 0], [0, 4], [0, 7]])
+        assert_array_almost_equal(out, xp.asarray([[0, 0], [0, 4], [0, 7]]))
         assert_array_almost_equal(out, ndimage.shift(data[:, ::2], (1, 1)))
 
     def test_map_coordinates_endianness_with_output_parameter(self):
         # output parameter given as array or dtype with either endianness
         # see issue #4127
-        data = np.array([[1, 2], [7, 6]])
-        expected = np.array([[0, 0], [0, 1]])
+        # NB: NumPy-only
```

Comment:
I'm not sure how this isn't causing errors - including `array_api_compatible` in `pytestmark` has required all tests to have an `xp` parameter in my experience; why isn't that the case here?

Elsewhere, we have used `skip_xp_backends(np_only=True)` to mark an `np`-only test - doing that everywhere makes things more maintainable overall, so please change to that here.

Reply:
> I'm not sure how this isn't causing errors - including array_api_compatible in pytestmark has required all tests to have an xp parameter in my experience; why isn't that the case here?

No errors:

```
$ python dev.py test -t scipy/ndimage/tests/test_interpolation.py -b all -v -- -k 'test_map_coordinates_endianness_with_output_parameter'
💻  ninja -C /home/ev-br/repos/scipy/build -j32
ninja: Entering directory `/home/ev-br/repos/scipy/build'
[2/2] Generating scipy/generate-version with a custom command
Build OK
💻  meson install -C build --only-changed --tags runtime,python-runtime,tests,devel
Installing, see meson-install.log...
Installation OK
SciPy from development installed path at: /home/ev-br/repos/scipy/build-install/lib/python3.11/site-packages
Running tests for scipy version:1.15.0.dev0+1198.4f80326, installed at:/home/ev-br/repos/scipy/build-install/lib/python3.11/site-packages/scipy
=========================================================================================== test session starts ===========================================================================================
platform linux -- Python 3.11.8, pytest-8.1.1, pluggy-1.4.0 -- /home/ev-br/.conda/envs/scipy-dev/bin/python
cachedir: .pytest_cache
hypothesis profile 'deterministic' -> database=None, deadline=None, print_blob=True, derandomize=True, suppress_health_check=[HealthCheck.data_too_large, HealthCheck.filter_too_much, HealthCheck.too_slow, HealthCheck.large_base_example, HealthCheck.function_scoped_fixture, HealthCheck.differing_executors]
rootdir: /home/ev-br/repos/scipy
configfile: pytest.ini
plugins: scipy_doctest-1.3, anyio-4.3.0, cov-4.1.0, timeout-2.2.0, hypothesis-6.99.2, xdist-3.5.0
collected 4940 items / 4935 deselected / 5 selected                                                                                                                                                       

scipy/ndimage/tests/test_interpolation.py::TestNdimageInterpolation::test_map_coordinates_endianness_with_output_parameter[numpy] PASSED                                                            [ 20%]
scipy/ndimage/tests/test_interpolation.py::TestNdimageInterpolation::test_map_coordinates_endianness_with_output_parameter[array_api_strict] PASSED                                                 [ 40%]
scipy/ndimage/tests/test_interpolation.py::TestNdimageInterpolation::test_map_coordinates_endianness_with_output_parameter[torch] PASSED                                                            [ 60%]
scipy/ndimage/tests/test_interpolation.py::TestNdimageInterpolation::test_map_coordinates_endianness_with_output_parameter[cupy] PASSED                                                             [ 80%]
scipy/ndimage/tests/test_interpolation.py::TestNdimageInterpolation::test_map_coordinates_endianness_with_output_parameter[jax.numpy] PASSED                                                        [100%]

=================================================================================== 5 passed, 4935 deselected in 0.33s ====================================================================================

```